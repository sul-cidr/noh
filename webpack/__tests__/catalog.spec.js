import fs from "fs";
import axios from "axios";
import mkdirp from "mkdirp";
import MockAdapter from "axios-mock-adapter";

import { main as catalogMain } from "../scripts/catalog";
import fixtures from "./__fixtures__/parser";

const mock = new MockAdapter(axios);

describe("parser", () => {
  let consoleError;
  let consoleInfo;
  let mkdirpSync;

  beforeEach(() => {
    mkdirpSync = mkdirp.sync;
    jest.spyOn(mkdirp, "sync").mockImplementation(() => {});
    consoleError = console.error;
    jest.spyOn(console, "error").mockImplementation(() => {});
    consoleInfo = console.info;
    jest.spyOn(console, "info").mockImplementation(() => {});
  });

  afterEach(() => {
    mkdirp.sync = mkdirpSync;
    console.error = consoleError;
    console.info = consoleInfo;
  });

  it("main raises an exception when no config is given", () => {
    expect(() => {
      catalogMain(null);
    }).toThrow();
  });

  it("main raises an exception when config is malformed", () => {
    const spy = jest
      .spyOn(fs, "readFileSync")
      .mockReturnValueOnce(JSON.stringify(0));
    expect(() => {
      catalogMain("path/to/config", true);
    }).toThrow();
    spy.mockRestore();
  });

  it("main downloads and parses the catalog of sections", (done) => {
    mock.reset();
    mock.onGet(fixtures.catalog.filters[0].url).reply(200, fixtures.catalogCSV);
    const spyRead = jest
      .spyOn(fs, "readFileSync")
      .mockReturnValueOnce(JSON.stringify(fixtures));
    const spyWrite = jest
      .spyOn(fs, "writeFileSync")
      .mockImplementation((file, jsonData) => {
        expect(jsonData).toMatchSnapshot();
      });
    catalogMain("path/to/config", false).then(() => {
      spyRead.mockRestore();
      spyWrite.mockRestore();
    });
    done();
  });
});
