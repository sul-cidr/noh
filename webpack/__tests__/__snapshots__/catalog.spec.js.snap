// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`parser main downloads and parses the catalog of sections 1`] = `
"{
  "filters": [
    {
      "by": "type",
      "values": [
        [
          "Spoken"
        ],
        [
          "Chanted",
          "Recitative chant"
        ],
        [
          "Chanted",
          "Introduction chant"
        ],
        [
          "Chanted",
          "Main chant"
        ],
        [
          "Chanted",
          "Closing chant"
        ],
        [
          "Entrance and exit music"
        ],
        [
          "Dance music"
        ]
      ]
    }
  ],
  "cards": [
    {
      "slug": "ageuta",
      "name": "Ageuta",
      "pills": [
        [
          "Chanted",
          "Main chant"
        ]
      ]
    },
    {
      "slug": "ashirai",
      "name": "Ashirai",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "hayafue",
      "name": "Hayafue",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "issei-chant",
      "name": "Issei chant",
      "pills": [
        [
          "Chanted",
          "Introduction chant"
        ]
      ]
    },
    {
      "slug": "issei-music",
      "name": "Issei music",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "jonomai",
      "name": "Jonomai",
      "pills": [
        [
          "Dance music"
        ]
      ]
    },
    {
      "slug": "kakaru",
      "name": "Kakaru",
      "pills": [
        [
          "Chanted",
          "Recitative chant"
        ]
      ]
    },
    {
      "slug": "kakeai",
      "name": "Kakeai",
      "pills": [
        [
          "Spoken"
        ]
      ]
    },
    {
      "slug": "kiri",
      "name": "Kiri",
      "pills": [
        [
          "Chanted",
          "Closing chant"
        ]
      ]
    },
    {
      "slug": "kuri",
      "name": "Kuri",
      "pills": [
        [
          "Chanted",
          "Introduction chant"
        ]
      ]
    },
    {
      "slug": "kuse",
      "name": "Kuse",
      "pills": [
        [
          "Chanted",
          "Main chant"
        ]
      ]
    },
    {
      "slug": "machiutai",
      "name": "Machiutai",
      "pills": [
        [
          "Chanted",
          "Introduction chant"
        ]
      ]
    },
    {
      "slug": "maibataraki",
      "name": "Maibataraki",
      "pills": [
        [
          "Dance music"
        ]
      ]
    },
    {
      "slug": "michiyuki",
      "name": "Michiyuki",
      "pills": [
        [
          "Chanted",
          "Introduction chant"
        ]
      ]
    },
    {
      "slug": "mondo",
      "name": "Mond≈ç",
      "pills": [
        [
          "Spoken"
        ]
      ]
    },
    {
      "slug": "nanori",
      "name": "Nanori",
      "pills": [
        [
          "Spoken"
        ]
      ]
    },
    {
      "slug": "nanoribue",
      "name": "Nanoribue",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "noriji",
      "name": "Noriji",
      "pills": [
        [
          "Chanted",
          "Closing chant"
        ]
      ]
    },
    {
      "slug": "notto-chant",
      "name": "Notto chant",
      "pills": [
        [
          "Chanted",
          "Recitative chant"
        ]
      ]
    },
    {
      "slug": "notto-music",
      "name": "Notto music",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "raijo",
      "name": "Raijo",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "rongi",
      "name": "Rongi",
      "pills": [
        [
          "Chanted",
          "Main chant"
        ]
      ]
    },
    {
      "slug": "sageuta",
      "name": "Sageuta",
      "pills": [
        [
          "Chanted",
          "Main chant"
        ]
      ]
    },
    {
      "slug": "sashi",
      "name": "Sashi",
      "pills": [
        [
          "Chanted",
          "Recitative chant"
        ]
      ]
    },
    {
      "slug": "shidai-chant",
      "name": "Shidai chant",
      "pills": [
        [
          "Chanted",
          "Introduction chant"
        ]
      ]
    },
    {
      "slug": "shidai-music",
      "name": "Shidai music",
      "pills": [
        [
          "Entrance and exit music"
        ]
      ]
    },
    {
      "slug": "tsukizerifu",
      "name": "Tsukizerifu",
      "pills": [
        [
          "Spoken"
        ]
      ]
    },
    {
      "slug": "waka",
      "name": "Waka",
      "pills": [
        [
          "Chanted",
          "Closing chant"
        ]
      ]
    }
  ]
}"
`;
